### DISCUSSION TITLE
ESP32 Helloworld Program Flash failed

### DISCUSSION THREAD

**Discussion by prasannabe2004** [Mair/esp32-course#97] [Course Qeustions] [answered]
Hi, I am trying to flash hello world program very first time to ESP32 bought from Amazon. while i run idf.py -p /dev/tty/USB0 flash monitor, it reports the below. I use my macbook, with ubuntu VM and attached the device to virtualbox for access.

Failed to connect to ESP32: Wrong boot mode detected (0x13)! The chip needs to be in download mode.

Any help is appreciated.

-Prasanna

**Comment by Mair** [comment]
hi @prasannabe2004 
It's a little hard to say without more info but it might be that some boards require you to hold down a button while attempting to flash.
Is there boot button on your device?

**Comment by prasannabe2004** [comment]
Thank you. Yes I tried to hold the Boot button while flashing, it works. Is that meant, any problem with the device?

**Comment by prasannabe2004** [comment]
Any workaround to avoid holding the boot button while flashing and monitor. It is not pretty consistent i have to repeat lot of times to get flashed and get into monitor mode.

**Reply by pbhereandnow** [reply]
any reply?

**Reply by prasannabe2004** [reply]
i tried another board. With that atleast it seems i need to hold button consistently.

**Comment by Mair** [comment, marked as answer]
@prasannabe2004,

usually, the boot pin is controlled by the DTR and RTS of the serial lines from the serial to USB converter. In other words, when the pc is ready to flash it "should" automatically change the pins voltage.
Have a look at this scematic
https://docs.zohopublic.com/file/kohvpddd1cff937ec4de5a828ac624e60a74d
in there you will notice the following
![image](https://user-images.githubusercontent.com/2477839/166611866-becf5d8b-b35c-4285-9d40-b70a62052577.png)

1. Check to see if you can find a schematic for your board
2. check if it has such a circuit
3. if it does then check on a scope if the DTR and RTS lines change when you try to flash

also maybe try a different cable or try it on a different PC if you can

Alternatively you may need another board and use that one as a spare

**Reply by mikebeepboop** [reply]
Not my question, but thanks for sharing that schematic - TIL something new.

**Reply by Mair** [reply]
@prasannabe2004 did you see above?

**Reply by Mair** [reply]
apologies @mikebeepboop 😁

**Comment by prasannabe2004** [comment]
@Mair - My board doesn't come with any schemantic. So not sure what it is. I tried with different PC and OS. No luck. So far still holding the boot button while flash and monitor as a work around. Though it is not convenient.

**Reply by Mair** [reply]
Unfortunate.
Might be worth while getting a second board

**Reply by noweare1** [reply]
Don't know if this helps but the command I use (for Ubuntu) is  idf.py -p /dev/ttyUSB0 flash monitor Not ...../dev/tty/USB0
